.github/workflows/cd.yml

```
@@ -1,18 +1,79 @@
 name: Continuous Deployment
 
 on:
+  workflow_dispatch:
+  schedule:
+    # At 00:00 on Saturday
+    # https://crontab.guru/#0_0_*_*_6
+    - cron: "0 0 * * 6"
   push:
     tags:
       - "v*.*.*"
 
+defaults:
+  run:
+    shell: bash
+
 jobs:
-  publish:
-    name: Publish on crates.io
+  publish-nightly:
+    name: Create a nightly release
     runs-on: ubuntu-latest
+    if: ${{ !startsWith(github.event.ref, 'refs/tags/v') }}
     steps:
       - name: Checkout the repository
         uses: actions/checkout@v3
-      - name: Publish
+        with:
+          fetch-depth: 0
+
+      - name: Calculate the next release
+        run: |
+          suffix="-alpha"
+          last_tag="$(git describe --abbrev=0 --tags `git rev-list --tags --max-count=1`)"
+          if [[ "${last_tag}" = *"${suffix}"* ]]; then
+            # increment the alpha version
+            alpha=$(echo "${last_tag}" | grep -oE '([0-9]+)$')
+            next_alpha=$((alpha + 1))
+            next_tag=$(echo "${last_tag}" | sed "s/\.[0-9]\+$/\.${next_alpha}/")
```

nit: no need for sed/grep here. [Bash is enough](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html).

```suggestion
          if [[ "${last_tag}" = *"-${suffix}"* ]]; then
            # increment the alpha version e.g. v0.22.1-alpha.12 -> v0.22.1-alpha.13
            alpha=${last_tag##*-${suffix}.}
            next_alpha=$((alpha + 1))
            next_tag=${last_tag/%${alpha}/${next_alpha}}
```